components/rmrk/Gallery/Gallery.vue

```diff
@@ -270,6 +274,13 @@ export default class Gallery extends mixins(PrefixMixin) {
     return params
   }
 
+  @Watch('$route.query.search')
+  protected onIdChange(val: string, oldVal: string) {
+    if (shouldUpdate(val, oldVal)) {
+      this.searchQuery.search = this.$route.query?.search?.toString() || ''
```

Comment:
Wouldn't it be better like this?
```suggestion
      this.searchQuery.search = val || ''
```